src/geo/transform.js

```diff
@@ -1661,7 +1687,16 @@ class Transform {
         // Z-axis uses pixel coordinates when globe mode is enabled
         const pixelsPerMeter = this.pixelsPerMeter;
 
-        this._projectionScaler = pixelsPerMeter / (mercatorZfromAltitude(1, this.center.lat) * this.worldSize);
+        if (this.projection.name !== 'globe') {
+            this._projectionScaler = pixelsPerMeter / (mercatorZfromAltitude(1, this.center.lat) * this.worldSize);
+        } else {
+            const refScale = mercatorZfromAltitude(1, 45) * this.worldSize;
+            const centerScale = mercatorZfromAltitude(1, this.center.lat) * this.worldSize;
+            const t = getProjectionInterpolationT(this, 1024);
+            const combinedScale = interpolate(refScale, centerScale, t);
+            this._projectionScaler = pixelsPerMeter / combinedScale;
+            this._mercatorScaleDifference = centerScale / refScale;
```

Comment:
Would `_mercatorScaleRatio` be a better name for this?

Reply:
Yep! done [71b67fb](https://github.com/mapbox/mapbox-gl-js/pull/11951/commits/71b67fbb81502e3b10ba9cb97ef1a237de0d7830)